snap.svg.zpd.js

```diff
@@ -403,50 +464,77 @@ SVGElement.prototype.getTransformToElement = SVGElement.prototype.getTransformTo
                     delta = event.detail / -9;       // Mozilla
                 }
 
-                var z = Math.pow(1 + zpdElement.options.zoomScale, delta);
+                _handleZoomingEvent(event, zpdElement, delta);
+            };
+            
+            var handleTouchMove = function handleTouchMove (event) {
 
-                var g = zpdElement.element.node;
+                if (!zpdElement.options.zoom) {
+                    return;
+                }
+
+                if (event.preventDefault) {
+                    event.preventDefault();
+                }
 
-                var p = _getEventPoint(event, zpdElement.data.svg);
+                if (!snapsvgzpd.enable) return;
 
-                p = p.matrixTransform(g.getCTM().inverse());
+                event.returnValue = false;
 
-                // Compute new scale matrix in current mouse position
-                var k = zpdElement.data.root.createSVGMatrix().translate(p.x, p.y).scale(z).translate(-p.x, -p.y);
+                // If multi-touch is true, then we are zooming instead of panning or dragging
+                if (zpdElement.data.touchZoom) {
 
-                _setCTM(g, g.getCTM().multiply(k), zpdElement.options.zoomThreshold);
+                    var distance = _getMultiTouchDistance(event);
 
-                if (typeof(stateTf) == 'undefined') {
-                    zpdElement.data.stateTf = g.getCTM().inverse();
-                }
+                    if (zpdElement.data.prevZoomDistance != null) {
+                        var delta = 0;
+
+                        if (zpdElement.data.prevZoomDistance > distance) delta = -0.1111111111111111;
+                        else delta = 0.1111111111111111;
```

Comment:
Yes the magic number, works well on iPad, I don't have many other touch devices to test it on.